### PR TITLE
Uml 1161 older lpa successful journey

### PR DESCRIPTION
**Pull request by GemTay**
# Purpose

Allows the user to request an activation key

Fixes UML-1161

## Approach

Hooked up UI pages to the API responses to complete older LPA journeys. Amended tests. Did lots of refactoring

## Checklist

* [ ] I have performed a self-review of my own code
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [ ] I have added tests to prove my work
* [ ] I have added welsh translation tags and updated translation files
* [ ] I have run an accessibility tool on any pages I have made changes to and fixed any issues found
* [ ] The product team have tested these changes

### CODE REVIEW

File: service-api/app/features/context/Acceptance/LpaContext.php [hunk 1]
```diff
@@ -44,9 +43,8 @@ public function iHaveBeenGivenAccessToUseAnLPAViaAPaperDocument()
         $this->userPostCode = 'string';
         $this->userFirstnames = 'Ian Deputy';
         $this->userSurname = 'Deputy';
-        // users dob is in this format when passed up from service-front
-        $this->userDob = '05/10/1975';
         $this->lpa->registrationDate = '2019-09-01';
+        $this->userDob = ['date' => '1975-10-05'];
```
Comment (GemTay) [inline service-api/app/features/context/Acceptance/LpaContext.php:47]: This is horrible and is work in progress. As I changed the front to pass the DOB as a datetime object, when the data is json_encoded for the request, it transforms it into an array to deal with in the API

File: service-front/app/features/actor-add-an-older-lpa.feature [hunk 2]
```diff
@@ -44,10 +134,9 @@ Feature: Add an older LPA
     And I confirm that those details are correct
     Then I am told that I cannot request an activation key
```
Comment (GemTay) [inline service-front/app/features/actor-add-an-older-lpa.feature:135]: @cooperaj I left this scenario without the UI test as to not cover your ticket too

File: service-front/app/features/actor-add-an-older-lpa.feature [hunk 3]
```diff
@@ -44,10 +134,9 @@ Feature: Add an older LPA
...
 
-  @integration
+  @ui @integration
   Scenario: The user cannot add an older LPA to their account if they have an activation key
     Given I am on the add an older LPA page
-    And I already have a valid activation key for my LPA
-    When I provide the details from a valid paper document
+    When I provide details from a valid LPA which I already have an activation key for
```
Comment (cooperaj) [inline service-front/app/features/actor-add-an-older-lpa.feature:140]: Am confused about the language change here? It also mixes fixture steps (And I...) with a doing step (When I...)

File: service-front/app/features/actor-add-an-older-lpa.feature [hunk 4]
```diff
@@ -9,28 +9,118 @@ Feature: Add an older LPA
...
 
-  @acceptance
-  Scenario: The user can access the appropriate triage page
+  @ui
+  Scenario: The user is taken to the add lpa triage page from the dashboard
     Given I am on the dashboard page
-    When I choose to add an LPA
-    And I do not have an activation key
-    Then I am asked for details from a paper LPA document
+    When I select to add an LPA
```
Comment (cooperaj) [inline service-front/app/features/actor-add-an-older-lpa.feature:15]: Why this choice of phrasing?

File: service-front/app/features/actor-add-lpa-triage.feature [hunk 5]
```diff
@@ -1,51 +0,0 @@
-@actor @addLpaTriage
```
Comment (cooperaj) [inline service-front/app/features/actor-add-lpa-triage.feature:1]: How come this has been merged? It makes for a much larger file?

File: service-front/app/features/context/Integration/LpaContext.php [hunk 6]
```diff
@@ -62,30 +63,12 @@ public function aLetterIsRequestedContainingAOneTimeUseCode()
...
-            ->respondWith(
-                new Response(
-                    StatusCodeInterface::STATUS_NO_CONTENT,
-                    [],
-                    ''
-                )
-            );
-
-        $lpaService = $this->container->get(LpaService::class);
+        $response = $this->lpaService->checkLPAMatchAndRequestLetter(
```
Comment (cooperaj) [inline service-front/app/features/context/Integration/LpaContext.php:66]: This file heavily touches on the PR i'm working on and have refactored it mostly already. As a general approach I've attempted to take the standard Given, When, Then scenario and implement the bulk of the work for the test in the final When. This should allow us to keep the files smaller by allowing reuse of steps that are the "same" but technically a little different.

File: service-front/app/features/context/Integration/LpaContext.php [hunk 7]
```diff
@@ -931,15 +855,52 @@ public function iProvideDetailsFromAnLPARegisteredBeforeSept2019()
      */
     public function iProvideDetailsThatDoNotMatchAValidPaperDocument()
     {
-        // Not needed for this context
+        // API call for getLpaById call happens inside of the check access codes handler
```
Comment (cooperaj) [inline service-front/app/features/context/Integration/LpaContext.php:858]: this being a case of that as it's a "When". The work for this scenario was mostly done in the following Then.

File: service-front/app/features/context/Integration/LpaContext.php [hunk 8]
```diff
@@ -1269,6 +1230,49 @@ public function theLPAIsSuccessfullyAdded()
         assertNotNull($actorCode);
     }
 
+    /**
+     * @When /^I provide details from a valid LPA which I already have an activation key for$/
```
Comment (cooperaj) [inline service-front/app/features/context/Integration/LpaContext.php:1234]:
```suggestion
     * @When /^I provide details of a valid LPA for which I already have an activation key$/
```

File: service-front/app/src/Actor/src/Handler/CheckYourAnswersHandler.php [hunk 9]
```diff
@@ -118,79 +117,43 @@ public function handleGet(ServerRequestInterface $request): ResponseInterface
...
-                        }
-                    }
-                }
-
-                //LPA check match and letter request sent
-                $twoWeeksFromNowDate = (new DateTime())->modify('+2 week');
-                return new HtmlResponse($this->renderer->render('actor::send-activation-key-confirmation', [
-                    'date' => $twoWeeksFromNowDate,
-                ]));
+        if ($this->form->isValid() && isset($this->data)) {
```
Comment (cooperaj) [inline service-front/app/src/Actor/src/Handler/CheckYourAnswersHandler.php:128]: Urgh.I'm currently neck deep in the refactor of this class (and LpaService). Getting the conflicts sorted for all this is going to suck.

File: service-front/app/features/actor-add-lpa-triage.feature (hunk 5 above)
Comment (GemTay) [reply service-front/app/features/actor-add-lpa-triage.feature:1]: Not sure why I did this originally, the file is now massive, i'll split it back out

File: service-front/app/src/Actor/src/Handler/CheckYourAnswersHandler.php (hunk 9 above)
Comment (GemTay) [reply service-front/app/src/Actor/src/Handler/CheckYourAnswersHandler.php]: I've reverted this so should be fine now

File: service-front/app/features/context/Integration/LpaContext.php (hunk 6 above)
Comment (GemTay) [reply service-front/app/features/context/Integration/LpaContext.php]: Reverted

**Pull request closed without merging**